roles/v2/mining-proxy/src/lib/downstream_mining.rs

```diff
@@ -419,22 +401,32 @@ impl
         &mut self,
         m: SubmitSharesStandard,
     ) -> Result<SendTo<UpstreamMiningNode>, Error> {
+        // TODO maybe we want to check if shares meet target before
+        // sending them upstream If that is the case it should be
+        // done by GroupChannel not here
         match &self.status {
             DownstreamMiningNodeStatus::Initializing => todo!(),
             DownstreamMiningNodeStatus::Paired(_) => todo!(),
             DownstreamMiningNodeStatus::ChannelOpened(Channel::DowntreamHomUpstreamGroup {
                 ..
             }) => {
                 let remote = self.upstream.as_ref().unwrap();
-                // TODO maybe we want to check if shares meet target before
-                // sending them upstream If that is the case it should be
-                // done by GroupChannel not here
                 let message = Mining::SubmitSharesStandard(m);
                 Ok(SendTo::RelayNewMessageToRemote(remote.clone(), message))
             }
+            DownstreamMiningNodeStatus::ChannelOpened(Channel::DowntreamHomUpstreamExtended {
+                ..
+            }) => {
+                // Safe unwrap is channel have been opened it means that the dowsntream is paired
+                // with an upstream
+                let remote = self.upstream.as_ref().unwrap();
+                let res = UpstreamMiningNode::handle_std_shr(remote.clone(), m).unwrap();
+                Ok(SendTo::Respond(res))
+            }
             _ => {
                 info!("Rceived share TODO send it somwhere");
```

Comment:
I'd love to understand more about when this happens. Can you write up a github issue around this panic so we can understand it and have a record to fix later.